docs/txp.md

```diff
@@ -0,0 +1,334 @@
+2. There is such input and `verifyAllIsKnown` is `False`. It means
+   that there was a protocol update not known to our software. In this
+   case we can't know whether that input is valid and how much value
+   it has. So we perform only basic
+   checks: [outputs checks](#outputs-checks)
+   and [consistency check](#consistency-check). Note that we don't
+   perform
+   [transaction's attributes check](#transaction-attributes-check),
+   because `verifyAllIsKnown` is `False`.
+3. All inputs have known format (i. e. `TxInUtxo`). In this case
+   (which should be the most common one) we can check all inputs'
+   legitimacy, compare input and output sums, check fee, etc. All
+   inputs must exist in UTXO. In the following we describe the case
+   when all inputs have `TxInUtxo` type and we assume that for each
+   input we know corresponding unspent output (if we don't,
+   transaction is invalid).
+
+All inputs of a transaction must be different and each input must be
+properly certified by its witness.
+
+### Witness checks
+
+Witness checks consist of two parts: we need to check that witness
+corresponds to the address from the output we want to spend and we
+need to check that witness itself is correct.
+
+* If witness is `PkWitness` (which contains a public key and a
+  signature), we require that address is a `PubKey` address and its
+  spending data is `PubKey` spending data with the same key as in
+  the witness. Witness itself is checked by checking validity of the
+  signature stored in this witness. Note: signature inside witness is
+  given for `TxSigData` (which is basically the same as `Hash Tx`)
+  with `SignTx` tag.
+* If witness is `ScriptWitness` (which contains validator and redeemer
+  scripts), we require that address is a `Script` address and its
+  spending data is `Script` spending data with validator script from
+  the witness. Witness itself is valid if few conditions are met.
+  * Both redeemer and validator script have the same version.
+  * If `verifyAllIsKnown` is `True`, script version (which is same for
+    redeemer and validator) must be known. Currently it means that
+    version must be equal to 0, it's the only version known to our software.
+  * Plutus validation script build from redeemer and validator for
+    given tx must return `True`.
+* If witness is `RedeemWitness` (which contains a redeem public key
+  and a redeem signature), we require that address is a `Redeem`
+  address and its spending data is `Redeem` spending data with the
+  same key as in the witness. Witness itself is checked in the same
+  way as `PkWitness` except that signature scheme is the one from
+  RSCoin and signing tag is `SignRedeemTx`.
+* If witness has unknown type with tag `t`, corresponding address must
+  have unknown type with tag `t`. Witness itself is valid only if
+  `verifyAllIsKnown` is `False`.
+
+Note: even though address doesn't contain its spending data, it's easy
```

Comment:
Not sure I follow, don't we miss info about that `Address'` contains attributes?

Reply:
Also, this note is cryptic and requires one to dive into code to be understood